File: farebot-app/src/main/java/com/codebutler/farebot/serialize/gson/CardTypeGsonTypeAdapter.java
package com.codebutler.farebot.serialize.gson;

import com.codebutler.farebot.card.CardType;
import com.google.gson.TypeAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;

import java.io.IOException;

public class CardTypeGsonTypeAdapter extends TypeAdapter<CardType> {

    @Override
    public void write(JsonWriter out, CardType value) throws IOException {
        out.value(value.name());
    }

    @Override
    public CardType read(JsonReader in) throws IOException {
        return CardType.valueOf(in.nextString());
    }
}

File: farebot-app/src/main/java/com/codebutler/farebot/app/core/serialize/gson/CardTypeGsonTypeAdapter.kt
/*
 * CardTypeGsonTypeAdapter.kt
 *
 * This file is part of FareBot.
 * Learn more at: https://codebutler.github.io/farebot/
 *
 * Copyright (C) 2017 Eric Butler <eric@codebutler.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package com.codebutler.farebot.app.core.serialize.gson

import com.codebutler.farebot.card.CardType
import com.google.gson.TypeAdapter
import com.google.gson.stream.JsonReader
import com.google.gson.stream.JsonWriter

class CardTypeGsonTypeAdapter : TypeAdapter<CardType>() {

    override fun write(out: JsonWriter, value: CardType) {
        out.value(value.name)
    }

    override fun read(`in`: JsonReader): CardType {
        return CardType.valueOf(`in`.nextString())
    }
}
